### PR TITLE
Adding terms of use in DDI

### LINKED ISSUE #9871: Feature Request/Idea: Add Standard CC and Custom Licenses to DDI Export Metadata
**Overview of the Feature Request**
We are using DDI XML exports from Dataverse in our Odesi data catalogue for searching and exploring datasets. We'd like to be able to display the Terms and Conditions as a button with a pop-up window containing the terms of use (including licenses) and terms of access stored in Dataverse's metadata for a given dataset before a user can download from the catalogue interface. 

See beta site: https://beta.odesi.ca fully connected to Dataverse repository in the backend

**What kind of user is the feature intended for?**
All users of DDI metadata coming from a Dataverse will benefit from having Dataverse's standard licenses (Terms of Use / Terms) be cross walked and made available in DDI Exports. Standard licenses such as Creative Commons CC-0 or CC-By are not mapped to DDI exports only the Terms of Access fields. 
See example DDI exports 1) [Dataverse Record example](https://borealisdata.ca/dataset.xhtml?persistentId=doi:10.5683/SP/AUCWOZ) 2) DDI export of that example (https://borealisdata.ca/api/datasets/export?exporter=ddi&persistentId=doi%3A10.5683/SP/AUCWOZ

**What inspired the request?**
we noticed that not all the terms metadata is being exported to DDI, especially CC licenses 

**What existing behavior do you want changed?**
Would be great to map this Dataverse terms metadata to DDI's 
dataAccs>setAvail/>useStmt/>dataAccs> section, like how it's done for when a Custom License is added to Dataverse
see: DDI export with custom license available (https://dataverse.harvard.edu/api/datasets/export?exporter=ddi&persistentId=doi%3A10.7910/DVN/ICISE4)

For example (custom license mapping to DDI from example linked above):
dataAccs>setAvail/>useStmt>
disclaimer>The <i>American Journal of Political Science</i> and the Odum Institute for Research in Social Science are not responsible for the accuracy or quality of data uploaded within the <i>AJPS</i> Dataverse, for the use of those data, or for interpretations or conclusions based on their use. /disclaimer>
useStmt>
dataAccs>

**Any brand new behavior do you want to add to Dataverse?**
Support for licenses in DDI 2.6 is a bit better than DDI 2.5 --- see https://ddi-alliance.atlassian.net/wiki/spaces/DDI4/pages/2851340301/Codebook+Tree+Diagram+Version+2.6

I'm thinking about this, but could confirm if 1.1.4.4 license (0-n) [URI, type] may work or there are additional fields supported under 2.6 dataAccs (0-n) has some new fields....TBD 

**Any open or closed issues related to this feature request?**
Yes this seems related a bit, in terms of the examples provided:

- https://github.com/IQSS/dataverse/issues/3648
- https://github.com/IQSS/dataverse/issues/8355

### PR DESCRIPTION
**Pull request by lubitchv**
**What this PR does / why we need it**:
It adds custom terms of use and standard license in DDI export.
It adds standard license to DDI import.
In this case DDI import and export match and will include both standard and custom licenses.

**Which issue(s) this PR closes**:

- Closes #9871 

**Suggestions on how to test this**:
Create two datasets, one with standard license and another custom license. Publish it. Look at export DDI. For standard license it should contain something like:
`<dataAccs>
<notes type="DVN:TOU" level="dv">
<a href="http://creativecommons.org/publicdomain/zero/1.0">CC0 1.0</a>
</notes>
</dataAccs>`
For custom license, it should contain something like:
`<dataAccs><notes type="DVN:TOU" level="dv">This is my terms</notes></dataAccs>`

Then try to create a new datasets with these DDI xmls using importddi api. See that for standard license it appears as standard and custom Terms of use appear as custom.

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java [hunk 1]
```diff
@@ -313,15 +315,36 @@ private static void writeDataAccess(XMLStreamWriter xmlw , DatasetVersionDTO ver
...
+
         /* any <note>s: */
+        if (version.getTermsOfUse() != null && !version.getTermsOfUse().trim().equals("")) {
+            xmlw.writeStartElement("notes");
+            xmlw.writeAttribute("type", NOTE_TYPE_TERMS_OF_USE);
+            xmlw.writeAttribute("level", LEVEL_DV);
+            xmlw.writeCharacters(version.getTermsOfUse());
+            xmlw.writeEndElement(); //notes
+        }
+
```
Comment (sekmiller) [inline src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java:327]: Can you add to DdiExportUtilTest.java to exercise this code?

File: src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java [hunk 2]
```diff
@@ -313,15 +315,36 @@ private static void writeDataAccess(XMLStreamWriter xmlw , DatasetVersionDTO ver
...
+            String name = license.getName();
+            String uri = license.getUri();
+            if ((name != null && !name.trim().equals("")) && (uri != null && !uri.trim().equals(""))) {
+                xmlw.writeStartElement("notes");
+                xmlw.writeAttribute("type", NOTE_TYPE_TERMS_OF_USE);
+                xmlw.writeAttribute("level", LEVEL_DV);
+                xmlw.writeCharacters("<a href=" + '"' + uri + '"' + ">" + name + "</a>");
+                xmlw.writeEndElement(); //notes
+            }
+        }
```
Comment (sekmiller) [inline src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java:347]: Also, please add a test to DdiExportUtilTest.java

**Review by sekmiller — CHANGES_REQUESTED** [review]
Looks good in general. Please add tests to DdiExportUtilTest.java and update from dev. We have recently released version 6.5.

Thanks!

File: src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java (hunk 2 above)
Comment (lubitchv) [reply src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java:347]: The standard license was already in dataset-finch1.json and dataset-finch1.xml was updated to add the standard license.

File: src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java (hunk 1 above)
Comment (lubitchv) [reply src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java:327]: Added dataset-finch-terms-of-use.json and dataset-finch-terms-of-use.xml which is used in testJson2DdiNoFilesTermsOfUse() for custom terms of use

**Comment by lubitchv** [comment]
@sekmiller I updated the branch from the develop and added to the DdiExportUtilTest to test for standard license and custom  terms of use

**Review by sekmiller — APPROVED** [review]
@lubitchv, thanks for the updates moving to ready for QA

**Comment by ofahimIQSS** [comment]
tested in internal - fix looks good. 

image
image

**Pull request merged**